src/engine.ts

```diff
@@ -31,11 +31,9 @@ export default class LuaEngine {
         // Contains the :await functionality.
         this.global.registerTypeExtension(1, createPromiseType(this.global, injectObjects))
 
-        if (injectObjects) {
```

Comment:
This should've remained in if you were doing a type extension imo. I'd have said you should inject the default null type extension always with a lower priority and then allow this to override it